ebpf_telemetry/ebpf_loader/ebpf_kern/ebpf_telemetry_kern_raw_tp.c

```diff
@@ -363,64 +372,92 @@ int sys_exit(struct bpf_raw_tracepoint_args *ctx)
+        //event->return_code = ctx->ret;
+        if (0 != bpf_probe_read(&event->return_code, sizeof(s64), (void *)&PT_REGS_RC(regs))){
+            BPF_PRINTK("ERROR, failed to get return code, exiting syscall %lu\n", event->syscall_id);
+            event->status = -1;
+        }
+        else{
+
+            // timestamp
+            event->bootns = bpf_ktime_get_ns();
+
+            // get the task struct
+            task = (void *)bpf_get_current_task();
+
+            // get the ppid
+            event->ppid = (u32)deref_ptr(task, config->ppid);
+
+            // get the session
+            event->auid = (u32)deref_ptr(task, config->auid);
+            event->ses = (u32)deref_ptr(task, config->ses);
+
+            if (!deref_string_into(event->tty, sizeof(event->tty), task, config->tty)){
+                bpf_probe_read_str(event->tty, sizeof(event->tty), notty);
+            }
+
+            // get the creds
+            cred = (void *)deref_ptr(task, config->cred);
+            if (cred) {
+                event->uid = (u32)deref_ptr(cred, config->cred_uid);
+                event->gid = (u32)deref_ptr(cred, config->cred_gid);
+                event->euid = (u32)deref_ptr(cred, config->cred_euid);
+                event->suid = (u32)deref_ptr(cred, config->cred_suid);
+                event->fsuid = (u32)deref_ptr(cred, config->cred_fsuid);
+                event->egid = (u32)deref_ptr(cred, config->cred_egid);
+                event->sgid = (u32)deref_ptr(cred, config->cred_sgid);
+                event->fsgid = (u32)deref_ptr(cred, config->cred_fsgid);
+            } else {
+                BPF_PRINTK("ERROR, ACCEPT failed to deref creds\n");
+                event->status = -1;
+
+                event->uid = -1;
+                event->gid = -1;
+                event->euid = -1;
+                event->suid = -1;
+                event->fsuid = -1;
+                event->egid = -1;
+                event->sgid = -1;
+                event->fsgid = -1;
+            }
+
+            // get the comm, etc
+            deref_string_into(event->comm, sizeof(event->comm), task, config->comm);
+            deref_filepath_into(event->exe, task, config->exe_path, config);
+            deref_filepath_into(event->pwd, task, config->pwd_path, config);
+
+            switch(event->syscall_id)
+            {
+                case __NR_accept4:
+                case __NR_accept:
+                {
+                    if (0 != bpf_probe_read(&event->data.socket.addr, 
+                                             sizeof(event->data.socket.addr), 
+                                             (void *)event->data.socket.addrp)){
+                        BPF_PRINTK("ERROR, ACCEPT failed to retrieve addr info\n");
+                        event->status = -1;
+                    }
+                    break;
+                }
+            }
+
+            // Pass the final result to user space if all is well
+            if (0 == event->status){
```

Comment:
Here I'm dropping the event if it fails to be complete based on the `event->status` code. Alternatively, we could report the "broken" event to user space.